### PR TITLE
Feature/filestorage sharenetworks create

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Share Network Create operation
For  #130

### CODE REVIEW

**Comment by ehdou** [comment]
Manila share network Create : https://github.com/openstack/manila/blob/master/api-ref/source/share-networks.inc#L162

**Comment by ehdou** [comment]
@jrperritt feel free to review if you have a moment

**Comment by jrperritt** [comment]
Thank you for following the style guide as closely as you could. One issue: this link (https://github.com/gophercloud/gophercloud/pull/118#issuecomment-253145324) looks like it's to documentation, not source code. In order to adequately review this PR (and indeed the others), we'll want a link to the function/schema/view from the source code to verify fields, types, return codes, etc. For example, many OpenStack services have a `views` directory for response objects, and some have a `schemas` directory for request body information. Still others have neither of those and just have logic defined in functions. As an example, have a look at the link in the 3rd bullet point of the [`On Pull Requests`](https://github.com/gophercloud/gophercloud/blob/master/STYLEGUIDE.md#on-pull-requests) section of the style guide.

**Comment by ehdou** [comment]
Thanks for your comment. I tried to find links in the code but it's not so simple. I found the [view](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/views/share_networks.py) and the [DB](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/db/sqlalchemy/models.py#L745). The [create](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/share_networks.py#L240) doesn't explicitly mention the return code, I'll try to find it. I also noticed that `name` and `description` are not required, I'll update my code in a moment.

**Comment by ehdou** [comment]
@jrperritt If I remove the test case for failing to build the request body (because `name` and `description` are not required parameters) then the coverage decreases -0.004%. How can I overcome this problem ? I don't see anything else I can test in my PR, except for `NewSharedFileSystemV2` that gets the Manila client, but then I would be the only one to test the client. Any suggestion ?

**Comment by ehdou** [comment]
It's a bit difficult to follow but response objects are assigned [here](https://github.com/openstack/manila/blob/master/manila/api/openstack/wsgi.py#L828). The default return code for response object is [200](https://github.com/openstack/manila/blob/master/manila/api/openstack/wsgi.py#L369).

**Comment by mvalkon** [comment]
@ehdou please replace #114 with #130 in the PR description.

File: openstack/sharedfilesystems/v2/sharenetworks/requests.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// these parameters, see the ShareNetwork object.
+type CreateOpts struct {
+	// The UUID of the Neutron network to set up for share servers
+	NeutronNetID string `json:"neutron_net_id,omitempty"`
+	// The UUID of the Neutron subnet to set up for share servers
+	NeutronSubnetID string `json:"neutron_subnet_id,omitempty"`
+	// The UUID of the nova network to set up for share servers
+	NovaNetID string `json:"nova_net_id,omitempty"`
+	// The share network name
+	Name string `json:"name"`
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharenetworks/requests.go]: `,omitempty`? Also for `Description` (below).

Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharenetworks/requests.go]: Actually, hold off on that. Let's merge this one and fix that later.

**Comment by jrperritt** [comment]
+2

**Pull request merged**